sdk/cognitiveservices/Knowledge.QnAMaker/src/Customizations/QnAMakerClient.cs

```diff
@@ -1,39 +1,35 @@
-﻿using Microsoft.Rest;
-using System;
-using System.Collections.Generic;
-using System.Linq;
-using System.Net.Http;
-using System.Text;
-using System.Threading.Tasks;
+﻿using System.Net.Http;
 
 namespace Microsoft.Azure.CognitiveServices.Knowledge.QnAMaker
 {
     public partial class QnAMakerClient
     {
         /// <summary>
-        /// Initializes a new instance of the QnAMakerClient class.
+        /// Initializes a new instance of the QnAMakerClient class for QnA Maker resources with hosted Runtime.
         /// </summary>
         /// <param name='credentials'>
         /// Required. Subscription credentials which uniquely identify client subscription.
         /// </param>
-        /// <param name="isPreview">
-        /// Optional. The flag used to change BaseUri if it is true.
-        /// </param>
-        /// <param name='rootHandler'>
-        /// Optional. The http client handler used to handle http transport.
-        /// </param>
         /// <param name='handlers'>
         /// Optional. The delegating handlers to add to the http client pipeline.
         /// </param>
         /// <exception cref="System.ArgumentNullException">
         /// Thrown when a required parameter is null
         /// </exception>
-        public QnAMakerClient(ServiceClientCredentials credentials, bool isPreview = false, params DelegatingHandler[] handlers) : this(credentials, handlers)
-        {
-            if (isPreview)
+        internal QnAMakerClient(EndpointKeyServiceClientCredentials credentials, params DelegatingHandler[] handlers) : this(handlers)
+        {            
+            if (credentials == null)
```

Comment:
This is a breaking change right, as you are now throwing an exception when the credentials are null?

And also, isn't the check below no longer correct since you will never get to line 27 because you are throwing the exception now?

Reply:
copied and pasted from generated similar constructor. Can fix this in customization. thanks.